### PR TITLE
Fix again

### PR DESCRIPTION
**Pull request by awebneck**
[Card](https://rentpath.atlassian.net/browse/SRV-2097)

There are more problems with the fs connector. This attempts fixing them in the following ways:
- Chronically uncaught error when seeking: Add the root hadoop FSError to the catch list around the `reader#seek` call in the `FsSourceTask` class.
- Batch Id changes not caught when there are no partial updates between bulk updates: Track the `isLast` value (only true when a given record is both the last in its file and its file is the last in its class [bulk/partial]) on the offset, and use this value to detect batch changes under the noted condition.
- Dupe records: Track "size" of an offset on the offset itself, and account for the stored value when performing seek at beginning of each polling cycle.
- Get rid of the confusing `skipLines` crap. We're not using it anymore, and it makes the logic gnarly to read.

### CODE REVIEW

**Comment by awebneck** [comment]
Closing due to confusing content. Will attempt again.

File: src/test/java/com/rentpath/kafka/connect/connectors/filesystem/readers/JsonFileReaderTest.java [hunk 1]
```diff
@@ -79,6 +80,16 @@ public void emptyFile() throws Throwable {
         super.emptyFile();
     }
 
+    @Test
+    public void testSingularLastFlag() {
```
Comment (sbrauer) [inline src/test/java/com/rentpath/kafka/connect/connectors/filesystem/readers/JsonFileReaderTest.java:84]: I don't see any assertions in here about a single last flag.
Is there a missing assertion or is the test misnamed?

File: src/main/java/com/github/mmolimar/kafka/connect/fs/policy/BulkIncrementalPolicy.java [hunk 2]
```diff
@@ -178,15 +179,20 @@ public FileMetadata extractExemplar(List<FileMetadata> batchFileMetadata) {
...
+        result.put(OFFSET_SIZE_OPT, offset.getRecordOffsetSize());
+        result.put(LAST_IN_CLASS, isLast);
         // The following values aren't needed for resuming reading where we left off.
         // Rather they are for use in appended metadata (for consumption by a downstream processor).
-        if ((priorOffset == null || !((Boolean) priorOffset.get(BULK_OPT)))
+        boolean precedingWasLastInClass = false;
+        if (priorOffset != null && priorOffset.get(LAST_IN_CLASS) != null)
+            precedingWasLastInClass = (Boolean) priorOffset.get(LAST_IN_CLASS);
+        if ((priorOffset == null || !((Boolean) priorOffset.get(BULK_OPT)) || precedingWasLastInClass)
                 && (Boolean) metadata.getOpt(BULK_OPT)) {
```
Comment (sbrauer) [inline src/main/java/com/github/mmolimar/kafka/connect/fs/policy/BulkIncrementalPolicy.java:196]: Would you tidy up the spacing here (the extra space before `&&`) or perhaps reformat/rewrite the conditional expression for clarity?
I wonder if it would be easier to grok (did I just say that?) if the order of the `&&` operands were swapped around, like:
```
if ((Boolean) metadata.getOpt(BULK_OPT) 
    && 
    (...)) 
```

File: src/test/java/com/rentpath/kafka/connect/connectors/filesystem/readers/JsonFileReaderTest.java (hunk 1 above)
Comment (awebneck) [reply src/test/java/com/rentpath/kafka/connect/connectors/filesystem/readers/JsonFileReaderTest.java:84]: Yeah, I need to remove this - it was an initial attempt at writing a test around the issue when I realized that problem was ultimately one that only occurred at the polling boundary, which would be difficult to simulate for the test.
